docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -90,4 +97,135 @@ hs.refreshIndex("empIndex", "incremental")
 TODO
 
 ### Hybrid Scan
-TODO
+
+Hybrid Scan enables to utilize existing index data along with newly appended source files or
+deleted existing files, without explicit refresh operation. For appended files, Hybrid Scan works
+by changing the query plan to perform on-the-fly shuffle of newly appended data and merge with
+the index data. For deleted files, Hyperspace also modify the plan to exclude the rows from deleted files
```

Comment:
Rephrase suggestion:
"For an index with appended source data files, HybridScan changes the query plan to shuffle new data on-the-fly and merge it with index records".